src/legacy/core_plugins/metrics/public/visualizations/views/timeseries/utils/series_styles.js

```diff
@@ -0,0 +1,67 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { CurveType } from '@elastic/charts';
+
+const DEFAULT_COLOR = '#000';
+
+export const getAreaStyles = ({ points, lines, color }) => ({
+  areaSeriesStyle: {
+    line: {
+      stroke: color,
+      strokeWidth: lines.lineWidth || 1,
+      visible: Boolean(lines.show),
+    },
+    area: {
+      fill: color,
+      opacity: lines.fill <= 0 ? 0 : lines.fill,
+      visible: Boolean(lines.show),
+    },
+    point: {
+      radius: points.radius || 0.5,
+      stroke: color || DEFAULT_COLOR,
+      strokeWidth: points.lineWidth || 5,
+      visible: points.lineWidth > 0 && Boolean(points.show),
+    },
+  },
+  curve: lines.steps ? CurveType.CURVE_STEP : CurveType.LINEAR,
+});
+
+export const getBarStyles = ({ show = true, lineWidth = 1, fill = 1 }, color) => ({
+  barSeriesStyle: {
+    border: {
+      stroke: color || DEFAULT_COLOR,
+      strokeWidth: lineWidth,
+      visible: show,
+    },
+    opacity: fill,
+  },
+});
+
+export const getSeriesColors = (color, specId) => {
+  const map = new Map();
+  /**
+   * TODO: colorValues for each column should be defined.
+   */
```

Comment:
Now it works w/o troubles but I was wondering how `colorValues` works under the hood because [the documentation](https://elastic.github.io/elastic-charts/?selectedKind=Stylings&selectedStory=custom%20series%20colors%20through%20spec%20props&full=0&addons=1&stories=1&panelRight=1&addonPanel=storybook%2Fstories%2Fstories-panel) describes it implicitly.

Reply:
The comment is removed